gateway/gateway_test.go

```diff
@@ -235,6 +235,35 @@ func TestUpdateReadings(t *testing.T) {
 	test.That(t, readings, test.ShouldResemble, expectedReadings)
 }
 
+func TestReadings(t *testing.T) {
+	g := &gateway{
+		lastReadings: make(map[string]interface{}),
+	}
+
+	// If lastReadings is empty and the call is not from data manager, return no error.
+	readings, err := g.Readings(context.Background(), nil)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, readings, test.ShouldResemble, map[string]interface{}{})
+
+	// If lastReadings is empty and the call is from data manager, return ErrNoCaptureToStore
+	readings, err = g.Readings(context.Background(), map[string]interface{}{data.FromDMString: true})
```

Comment:
[opt/q] is the behavior the same if you have `data.FromDMString: false`? just curious with how interfaces work

Reply:
no error should be returned if the FromDMString is false, added a test case for that.